src/connection/transport/sasl.rs

```diff
@@ -0,0 +1,25 @@
+#[derive(Debug, Clone)]
+pub enum SaslConfig {
+    Plain { username: String, password: String },
+}
+
+impl SaslConfig {
+    pub fn auth_bytes(&self) -> Vec<u8> {
+        match self {
+            Self::Plain { username, password } => {
+                let mut auth: Vec<u8> = Vec::new();
+                auth.push(0);
+                auth.extend(username.bytes());
+                auth.push(0);
+                auth.extend(password.bytes());
```

Comment:
Can you add a link/reference for this binary format?

I think for `PLAIN` it's OK to hand-roll this, in the future however I would like to offload this to another crate because I don't wanna deal w/ all the SASL crypto stuff in rskafka.

Reply:
Are you okay depending on a pure rust crate such as this one? https://docs.rs/sasl/latest/sasl/
The other SASL packages are rust wrappers around C implementations.